ExampleMod/Common/ExamplePlayerDrawLayer.cs

```diff
@@ -0,0 +1,57 @@
+using ExampleMod.Content.Items;
+using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using ReLogic.Content;
+using Terraria;
+using Terraria.DataStructures;
+using Terraria.ModLoader;
+
+namespace ExampleMod.Common
+{
+	public class ExamplePlayerDrawLayer : PlayerDrawLayer
+	{
+		private Asset<Texture2D> exampleItemTexture;
+
+		//Returning true in this property makes this layer appear on the minimap player head icon.
+		public override bool IsHeadLayer => true;
+
+		public override bool GetDefaultVisiblity(PlayerDrawSet drawInfo) {
+			//The layer will be visible only if the player is holding an ExampleItem in their hands. Or if another modder forces this layer to be visible.
+			return drawInfo.drawPlayer.HeldItem?.type == ModContent.ItemType<ExampleItem>();
+
+			// If you'd like to reference another PlayerDrawLayer's visibility,
+			// you can do so by getting its instance via ModContent.GetInstance<OtherDrawLayer>(), and calling GetDefaultVisiblity on it
+		}
+
+		// This layer will be a 'child' of the head layer, and draw before (beneath) it.
+		// If the Head layer is hidden, this layer will also be hidden.
+		// If the Head layer is moved, this layer will move with it.
+		public override Position GetDefaultPosition() => new Before(PlayerDrawLayers.Head);
+		// If you want to make a layer which isn't a child of another layer, use `new Between(Layer1, Layer2)` to specify the position.
+		// If you want to make a 'mobile' layer which can render in different locations depending on the drawInfo, use a `Multiple` position.
+
+		protected override void Draw(ref PlayerDrawSet drawInfo) {
+			//The following code draws ExampleItem's texture behind the player's head.
+
+			if (exampleItemTexture == null) {
+				exampleItemTexture = ModContent.GetTexture("ExampleMod/Content/Items/ExampleItem");
```

Comment:
It would be awesome if assets could be lazy loaded, just define the path in the constructor when you define the asset.

Reply:
They will be autoloaded, and this code doesn't conflict with that, since it returns an asset, which doesn't have to be loaded immediately.

Reply:
I don't mean autoloaded, I mean lazy loaded. That means, you would do something like:
`Asset myAsset = new Asset<Texture2D>("My/Path/To/Asset");`
The null check and loading GetTexture I commented on essentially is the lazy loading inside the asset you define. 
This just gets rids of null checks and calling GetTexture all the time for your assets.

Reply:
See also: https://docs.microsoft.com/en-us/dotnet/framework/performance/lazy-initialization

Reply:
I did a typo. I meant that we plan to make assets load their .Values on-demand.